docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,310 @@ title: KeptnTaskDefinition
+    for more information about the runtime.
+
+    The `function` can be defined as one of the following:
+
+    * **inline** - Include the actual executable code to execute.
+      This can be written as a full-fledged Deno script
+      that is included in this file.
+      For example:
+
+      ```yaml
+      function:
+        inline:
+          code: |
+            console.log("Deployment Task has been executed");
+      ```
+
+    * **httpRef** - Specify a Deno script to be executed at runtime
+      from the remote webserver that is specified.
+      For example:
+
+      ```yaml
+      name: hello-keptn-http
+        spec:
+            function:
+              httpRef:
+                url: "https://www.example.com/yourscript.js"
+      ```
+
+      Note that the file referenced is actually JavaScript,
+      which is essentially the same as a Deno script.
+
+    * **functionRef** -- Execute another `KeptnTaskDefinition` that has been defined.
+      Populate this field with the value of the `name` field
+      for the `KeptnTaskDefinition` to be called.
+      This is commonly used to call a general function
+      that is used in multiple places, possibly with different parameters.
+      An example is:
+
+       ```yaml
+       spec:
+         function:
+           functionRef:
+             name: slack-notification
+       ```
+
+      This can also be used to group a set of tasks
+      into a single `KeptnTaskDefinition`,
+      such as defining a `KeptnTaskDefinition` for testing.
+      In this case, it other, existing `KeptnTaskDefinition`s
+      for each type of test to be run,
+      specifying each by the value of the `name` field.
+
+  * **parameters** - An optional field to supply input parameters to a function.
+    The Lifecycle Toolkit passes the values defined inside the `map` field
+    as a JSON object.
+    For example:
+
+     ```yaml
+       spec:
+         parameters:
+           map:
+             textMessage: "This is my configuration"
+     ```
+
+     The JSON object can be read
+     through the `DATA` environment variable using `Deno.env.get("DATA");`.
+
+     Multi-level maps are not supported at this time.
+
+     Currently only one secret can be passed.
+     The secret must have a `key` called `SECURE_DATA`.
+     It can be accessed via the environment variable `Deno.env.get("SECURE_DATA")`.
+     See [Context](#context) for details.
+
+  * **secureParameters** -- An optional field used to pass a Kubernetes secret.
+    The `secret` value is the Kubernetes secret name
+    that is mounted into the runtime and made available to functions
+    using the `SECURE_DATA` environment variable.
+    For example:
+
+    ```yaml
+    secureParameters:
+      secret: slack-token
+    ```
+
+    See [Create secret text](#create-secret-text) for details.
+
+## Usage
+
+A Task is responsible for executing the TaskDefinition of a
+[workload](https://kubernetes.io/docs/concepts/workloads/)..
```

Comment:
Here we can say, that workload or application, as on both levels, the principles are the same